src/lib.rs

```diff
@@ -26,11 +26,13 @@ mod common;
 pub mod concept;
 mod connection;
 mod database;
+pub mod logic;
```

Comment:
`logic` pretty clearly belongs under `transaction` next to `query` and `concept`.

Reply:
Actually, we should have two `logic` modules, come to think, siumilar to how we deal with `concept`: `logic::Rule`, which is a POD struct, and `transaction::logic::LogicManager`.